monai/transforms/inverse.py

```diff
@@ -89,8 +90,12 @@ def push_transform(
 
     def check_transforms_match(self, transform: dict) -> None:
         """Check transforms are of same instance."""
-        if transform[InverseKeys.ID.value] != id(self):
-            raise RuntimeError("Should inverse most recently applied invertible transform first")
+        if transform[InverseKeys.ID.value] == id(self):
+            return
+        # basic check if windows because of multiprocessing differences (objects get recreated so don't have same ID)
+        if sys.platform == "win32" and transform[InverseKeys.CLASS_NAME.value] == self.__class__.__name__:
+            return
```

Comment:
more specific check (need `import torch`):
```suggestion
        # basic check if multiprocessing uses 'spawn' (objects get recreated so don't have same ID)
        if (
            torch.multiprocessing.get_start_method(allow_none=True) == "spawn"
            and transform[InverseKeys.CLASS_NAME.value] == self.__class__.__name__
        ):
            return
```